### ISSUE TITLE
add fine-tune cli in scripts

### ISSUE THREAD

**Issue by fvarno** [varnio/fedsim#6]
A fine-tune cli option is needed to tune the algorithms and return the best value. Skopt seems a safe choice (?). The values that are passed through as optional algorithm arguments (--a-) are those that should be tunes. It would be helpful to get these arguemnts as a range (such that the way space of search is defined in skopt).

**Comment by fvarno** [comment]
We already changed the argument delivering from `--a-<arg> value` form to arg `arg:value` in front of the object name so the plan for adding fine-tuning should have some changes. Also we need fine-tuning for global arguments such as learning rate.

**Comment by fvarno** [comment]
It seems that the easiest way to handle fine-tuning for both the general hyper-parmeters and object ones (e.g., mu in algorithm class), it is better to have a fine_tune method defined for the algorithm class.
